### PR TITLE
Enhance : New Total Solar eclipse added

### LINKED ISSUE #274: Add a new events to time travel
" Is it possible to add the April 8, 2024 Total Solar Eclipse to the Time Travel list?
" - user feedback.

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #274 

- [x] Run the unit tests with `gradlew app:connectedGmsDebugAndroidTest` to make sure you didn't break anything (with an emulator running or phone connected)

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by jddeep** [comment]
@jaydeetay Please review.

**Comment by utkarsh530** [comment]
@jddeep I think its a good practice to let first the mentor assign the task to some person and wait. I can see that some person has already requested before in #274 . Please avoid this.

**Comment by jddeep** [comment]
> @jddeep I think its a good practice to let first the mentor assign the task to some person and wait. I can see that some person has already requested before in #274 . Please avoid this.

@utkarsh530 I am a strong believer of what you said here. But as you might have seen this issue was an enhancement and was a user feedback. So I thought it would be better to directly make a PR and let it be reviewed. Because if we would have waited to get assigned for every issue the development process would have taken too long. But nevertheless It is always a good practice to get assigned first. But in this case I had to make an exception. There are lots of aspects to be worked on. Hope you understand my point buddy. Cheers!

**Comment by aayushsingla** [comment]
@jddeep this doesn't make sense at all. If you wanted to work on this then you should have asked me first whether if I'm working on it or not. I was already working on this issue and would have sent PR by tonight.

**Comment by jddeep** [comment]
> @jddeep this doesn't make sense at all. If you wanted to work on this then you should have asked me first whether if I'm working on it or not. I was already working on this issue and would have sent PR by tonight.

@aayushsingla If you had commented on the issue that you will be working on it instead of asking for it. I would have surely made sure of that. But here I think there has been a lack of miscommunication as well. Well in that case I would really recommend you to let it be in a branch and do some more work in another one. And let it be untill this PR get's reviewed. Then after that you can make a PR from your side. Untill then I really apologise for the miscommunication. Hope you understand this time buddy.

**Comment by jddeep** [comment]
Till then feel free to work on some other issue. Cheers!

**Comment by jddeep** [comment]
@aayushsingla If I had seen an affirmation that you would be working on it, I would surely have left that, But as the issue had been given an Enhancement, I decided to directly make a PR and then get it reviewed at once instead of waiting to get assigned. You see if we wait to get assigned for every issue, it would extend the development process. But from now on let us all communicate on gitter and help each other in development. Cheers!

**Comment by jddeep** [comment]
@jaydeetay Please review this,

**Comment by jaydeetay** [comment]
Hi folks. Thanks for the PR. As you'll have figured out by now, I don't have enough time to police who should be working on what. I wish I did, as clearly you guys have lots of enthusiasm and energy and I'm certainly flattered that you're putting it into Sky Map. So, I'm sorry that you've ended up treading on each others' toes here. It happens. Generally speaking I'm going to examine PRs in the order that they arrive in my inbox (very slowly, probably). As always, follow the README and check in with me before embarking on significant feature work.

File: app/src/main/java/com/google/android/stardroid/activities/dialogs/TimeTravelDialog.java [hunk 1]
```diff
@@ -244,6 +244,9 @@ private void setPopularDate(int popularDateIndex) {
       case 8:  // Moon Landing 1969.
         setDate(new Date(-14182953622L));
         break;
+      case 9:
+        setDate(new Date(1712534400));
```
Comment (jaydeetay) [inline app/src/main/java/com/google/android/stardroid/activities/dialogs/TimeTravelDialog.java:248]: Is this date correct? It has many fewer digits than the others.

Comment (jddeep) [reply app/src/main/java/com/google/android/stardroid/activities/dialogs/TimeTravelDialog.java:248]: Well @jaydeetay I used the time stamp date converter as the source. I myself don't know how to calculate these date and times for be honest!

**Comment by jaydeetay** [comment]
I'm not convinced the date is right here.

**Pull request closed without merging**